File: Python/insertion_sort.py
def insertion_sort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1
        while j >= 0 and key < arr[j]:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = key
    return arr

def bucket_sort(arr):
    max_val = max(arr)
    size = max_val // len(arr)
    buckets = [[] for _ in range(len(arr))]
    for num in arr:
        index = num // size
        if index != len(arr):
            buckets[index].append(num)
        else:
            buckets[len(arr) - 1].append(num)
    for bucket in buckets:
        bucket=insertion_sort(bucket)
    result = []
    for bucket in buckets:
        result.extend(bucket)
    return result

File: C#/InsertionSort.cs
using System;

class InsertionSort
{
    public static void Sort(int[] arr)
    {
        int n = arr.Length;
        for(int i = 1; i < n; i++)
        {
            int key = arr[i];
            int j = i - 1;

            while(j >= 0 && arr[j] > key)
            {
                arr[j +1] = arr[j];
                j--;
            }
            arr[j + 1] = key;
        }
    }
    
    public static void Main()
    {
        int[] arr = { 12, 11, 13, 5, 6 };
        Sort(arr);
        Console.WriteLine(string.Join(" ", arr));
    }
}